repository-data/webfiles/src/main/resources/site/freemarker/common/supplementary-information-hub-article.ftl

```diff
@@ -0,0 +1,145 @@
+<#ftl output_format="HTML">
+<#include "../include/imports.ftl">
+<#include "macro/stickyNavSections.ftl">
+<#include "macro/stickyNavYears.ftl">
+<#include "macro/stickyNavTags.ftl">
+<#include "macro/hubBox.ftl">
+<#include "macro/documentHeader.ftl">
+<#include "macro/stickyGroupBlockHeader.ftl">
+<#include "macro/stickyNavYears.ftl">
+
+<@hst.setBundle basename="rb.generic.headers,rb.generic.labels,rb.doctype.supplementary-info-hub "/>
+
+<#-- Add meta tags -->
+<#include "macro/metaTags.ftl">
+<@metaTags></@metaTags>
+
+<#assign monthNames = monthsOfTheYear()?reverse />
+
+<#--Group the documents by month  -->
+<#assign eventGroupHash = {} />
+<#list pageable.items as item>
+    <#if item.publishedDate?size gt 0>
+        <@fmt.formatDate value=item.publishedDate.time type="Date" pattern="MMMM" timeZone="${getTimeZone()}" var="key" />
+        <#assign eventGroupHash = eventGroupHash + {  key : (eventGroupHash[key]![]) + [ item ] } />
+    </#if>
+</#list>
+
+<#-- Return the filter navigation links for the year -->
+<#function getFilterYearLinks>
+    <#assign links = [] />
+
+    <#list years as key>
+        <#assign typeCount = years?size />
+        <#assign links += [{ "key" : key, "title": key }] />
+    </#list>
+
+    <#return links />
+</#function>
+
+<#assign hasIntroductionContent = document.introduction.content?has_content />
+
+<article class="article article--supplementary-info-hub">
+    <@documentHeader document 'supplementary-info-hub' null null null null false></@documentHeader>
```

Comment:
The interface of `documentHeader` with all those `null` values, does not seem very tidy. Can 'documentHeader' be reworked to do something like take in an options object/array/map?

Reply:
The `@documentHeader` macro is used in 13 templates, so I think it would be a fair bit of work to change the list of fields into an object/hash. I went for using `null` values for those fields, but as I see that other templates (such as https://github.com/NHS-digital-website/hippo/blob/DW-1163-create-supplementary-info-page/repository-data/webfiles/src/main/resources/site/freemarker/common/service/service-article.ftl#L40) use `''` to pass in a value is not needed or available. I think as a minimum I should change `null` to `''` to stick to the existing coding conventions, but I'm not sure how much effort is required to refactor the fields into an object. :man-shrugging:

Reply:
Yes, I see the `@documentHeader` evolved over time. Using `''` will be okay.

Reply:
Sure, I'll do the update from `null` -> `''`.

Reply:
@josephcyrus-valtech just updated the `@docHeader`, so it uses empty strings as per the `service-article` template.